File: 250829/최고의 33위치/best-place-of-33.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[][] grid = new int[n][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                grid[i][j] = sc.nextInt();
            }
        }
        // Please write your code here.

        int totalMax = 0;

        for (int i = 0; i < n - 2; i++) {
            for (int j = 0; j < n - 2; j++) {
                int count = 0;
                for (int a = 0; a < 3; a++) {
                    for (int b = 0; b < 3; b++) {
                        if (grid[i + a][j + b] == 1) {
                            count++;
                        }
                    }
                }
                totalMax = Math.max(totalMax, count);
            }
        }

        System.out.println(totalMax);
    }
}

File: 260312/최고의 33위치/best-place-of-33.py
n = int(input())
grid = [list(map(int, input().split())) for _ in range(n)]

# Please write your code here.

dx = [
    -1, -1, -1,
    0, 0, 0,
    1, 1, 1
]

dy = [
    -1, 0, 1,
    -1, 0, 1,
    -1, 0, 1
]

max_sum = 0
for r in range(n):
    for c in range(n):
        sum = 0
        for i in range(9):
            xx = r + dx[i]
            yy = c + dy[i]

            if 0 <= xx and xx < n and 0 <= yy and yy < n:                
                sum += grid[xx][yy]
    
        max_sum = max(max_sum, sum)

print(max_sum)

